LDMP/models/algorithms_delegate.py

```diff
@@ -0,0 +1,263 @@
+    "... x ssss x ...",
+    ".... x ss x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x  ss  x ...",
+    ".. x        x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+_icon2_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x        x ..",
+    "... x ssss x ...",
+    ".... x ss x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x  ss  x ...",
+    ".. x   ss   x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+_icon3_xpm = [
+    "16 16 4 1",
+    ".  c None",
+    "s  c #cccc00",
+    "x  c #ffffff",
+    "   c #000000",
+    "................",
+    ".              .",
+    ". xxxxxxxxxxxx .",
+    ". x          x .",
+    ".. x        x ..",
+    "... x      x ...",
+    ".... x    x ....",
+    "..... x  x .....",
+    "..... x  x .....",
+    ".... x ss x ....",
+    "... x ssss x ...",
+    ".. x ssssss x ..",
+    ". x          x .",
+    ". xxxxxxxxxxxx .",
+    ".              .",
+    "................"]
+
+class AlgorithmItemDelegate(QStyledItemDelegate):
+
+    needsRedraw = pyqtSignal()
+
+    def __init__(self, parent: QObject = None):
+        super().__init__(parent)
+
+        # create widget but do not display it. It will be used to grab it during paint
+        # In this way is efficient and not rendered the widget every paint envet.
+        # Working widget is rendered and displayed only during createEditor of the cell
+        # in this way the widget is represented, but active only during editing role.
+        # This speedup View rendering.
+        # observe that I just create the top level ToolButton because is that 
+        # statically rendered
+        # self.runAction = QAction(tr('Run'))
+        # self.editorWidget = QToolButton()
+        # self.editorWidget.setPopupMode(QToolButton.MenuButtonPopup)
+        # self.editorWidget.setMenu(QMenu())
+        # self.editorWidget.menu().addAction(self.runAction)
+        # self.editorWidget.setDefaultAction(self.runAction)
+
+        # pixmap animation control
+        self.current = 0
+        self.timerId = self.startTimer(250)
+        self.pixmaps = (QPixmap(_icon0_xpm),
+                        QPixmap(_icon1_xpm),
+                        QPixmap(_icon2_xpm),
+                        QPixmap(_icon3_xpm))
+        self.needsRedraw.connect(parent.viewport().update)
+
+        # create svg renderer
+        # self.svg_renderer = QSvgRenderer('/home/ginetto/Downloads/tcjp3h.svg', parent)
+        # self.svg_renderer = QSvgRenderer('/home/ginetto/Downloads/spinner-of-dots-svgrepo-com.svg', parent)
+        # self.svg_renderer.repaintNeeded.connect(parent.viewport().update) 
```

Comment:
Please remove this if it is not being used, as it creates clutter

Reply:
this is placeholder just to remember that is possible to add a dynamic svg probably useful for the Dataset tree view part... please leave there temporarily to until the end of datasets tab implementation